src/analysis/special_functions/pow.lean

```diff
@@ -1442,3 +1433,105 @@ begin
 end
 
 end ennreal
+
+section measurability_ennreal
+
+lemma ennreal.measurable_rpow : measurable (λ p : ennreal × ℝ, p.1 ^ p.2) :=
```

Comment:
This is a long proof, I think there are some lemmas in the library you can use to shorten it.

I think you can consider the `top` case separately using `ennreal.measurable_of_measurable_nnreal_prod`, and afterwards the `0` case using `measurable_of_measurable_on_compl_singleton`. That should shorten the proof significantly.

Reply:
Oh wait, `measurable_of_measurable_on_compl_singleton` doesn't work. Is there another lemma you can separate out that makes it easier to handle the `0` case? Or does `measurable_of_measurable_union_cover` do exactly what you want?

At least `ennreal.measurable_of_measurable_nnreal_prod` should be a big improvement, so that you don't have to work with `{p : ennreal × ℝ | p.1 ≠ ⊤}` can can work with `nnreal × ℝ` instead.

Reply:
I wanted to use measurable_of_measurable_on_compl_singleton but it indeed does not work becaus I want to remove {0}*univ, so I used measurable_of_measurable_union_cover to do that.
Thanks for telling me about ennreal.measurable_of_measurable_nnreal_prod, I did not know it existed. I'll see what I can do with it.

Reply:
Looks like I will have the same three cases, but with notably lighter notation.

Reply:
Why do you actually need the `0` case? Doesn't the `nnreal × ℝ` case follow from `real.measurable_rpow`?

Reply:
I need the 0 case because I use `ennreal.coe_rpow_of_ne_zero` to go from `(\u(x:nnreal) ^ y)` to `\u(x ^y)`, and then obtain the measurability of the composition of coe and the nnreal version of rpow .

Something that may be related to that problem is the fact that 0 ^ x is equal to top for x < 0 in ennreal, but is defined as 0 for nnreal.

I have not looked at the possibility of going through reals instead of nnreals yet.

Reply:
Oh, if the exponentials aren't defined the same on 0, then you indeed need to do 0 separately (or case on the exponent, but I doubt that will be shorter).

Reply:
The proof looks much better now.

Looking over the proof, I still feel like all steps that involve subtypes are a detour. After applying `ennreal.measurable_of_measurable_nnreal_prod`, you can rewrite with the equation
```lean
x ^ y = 
  if x = 0 ∧ y < 0 then ⊤ else (x ^ y : ℝ≥0)
``` 
(which is true by definition, but maybe also should be a lemma), and then apply `measurable.ite` directly, right?